### ISSUE TITLE
:error_logger.add_report_handler(Sentry.Logger) not working on remote_console (production)

### ISSUE THREAD

**Issue by lorenzosinisi** [getsentry/sentry-elixir#260]
### Environment

* Elixir version (elixir -v):  1.5.1
* Erlang/OTP version (erl): Erlang/OTP 19
* Sentry version (mix deps): 6.0.5
* Operating system: Linux

### Description
Hello, thanks for the great job on the Sentry client but I have found an issue. When I configure Sentry by adding also :error_logger.add_report_handler(Sentry.Logger) in the `start` function in the following way: 

```
   ... 
    res = Supervisor.start_link(children, opts)

    if !(Sentry.Logger in :gen_event.which_handlers(:error_logger)) do
      :ok = :error_logger.add_report_handler(Sentry.Logger)
    end

    res
  end
```

The runtime exceptions are not reported while when I explicitly capture them (using the Sentry library) they are reported. 

If I run in the console `Sentry.capture_exception("test", [stacktrace: System.stacktrace(), extra: %{extra: "test"}])`, this get's sent to Sentry (in the remote machine as well as when I run the app in development) 

### Expected Behavior

When I log in production and open the remote_console, I run ``Task.async(fn -> raise("ooops") end)``` and I receive a Sentry notification about an Erlang error that just occurred. 

### Actual Behavior
When I log in production and open the remote_console, I run ``Task.async(fn -> raise("ooops") end)``` nothing happens. 

### Steps to Reproduce

ssh on your production or staging machine configuring Sentry to capture the current env, open the remote console with `./bin/nameofyourapp remote_console` and run ``Task.async(fn -> raise("ooops") end)```

Anybody has any idea on how to debug that? Is it maybe that in the remote console runtime errors don't get reported? I am worried that if the app crashed in production I would not get any notification. Also, is there anything that I can check to make sure that the events are actually sent to the Sentry.Logger gen_event?

**Comment by lorenzosinisi** [comment]
`:gen_event.which_handlers :error_logger # => [Sentry.Logger, :sasl_report_tty_h, Logger.ErrorHandler, :error_logger]` is what I see when I run this code on prod :/

**Comment by mitchellhenke** [comment]
@lorenzosinisi thanks for opening an issue, I'll definitely look into this.

Are you building releases using [distillery](https://hexdocs.pm/distillery)?

**Comment by lorenzosinisi** [comment]
Yes, using distillery. I was inspecting the problem and I found out that if you are in the remote console and send a message that breaks one of the GenServer modules currently running, you do get a report. But not if you create an error only in the console. Is definitely has to do with the way the remote_console works

**Comment by mitchellhenke** [comment]
I took a look into this, and I see the issue, but I'm not sure how to go about solving it yet.  It's hitting this clause: https://github.com/getsentry/sentry-elixir/blob/master/lib/sentry/logger.ex#L46-L48, but Elixir's `Logger.ErrorHandler` has the same [clause](https://github.com/elixir-lang/elixir/blob/master/lib/logger/lib/logger/error_handler.ex#L25-L27) and still gets called correctly.

I'll have to try to find some more information on why that would be.

**Comment by mitchellhenke** [comment]
@lorenzosinisi would you be able to give the master branch a try and see if the incorrect behavior persists?
